### PR TITLE
A bunch of fixes for basic-install.sh

### PR DESCRIPTION
**Pull request by cryptomilk**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**

This PR provides several cleanup commits for the `basic-install.sh` script.

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -859,6 +863,10 @@ setStaticIPv4() {
     # Local, named variables
     local IFCFG_FILE
     local CONNECTION_NAME
+    # If a static interface is already configured, we are done.
+    if [[ -r /etc/sysconfig/network/ifcfg-${PIHOLE_INTERFACE} ]]; then
```
Comment (dschaper) [inline automated install/basic-install.sh]: Does this guarantee that the interface is set with a static IP address, or just that the interface exists and may still be assigned a dynamic address?

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -1638,23 +1647,23 @@ installPiholeWeb() {
     local str="Creating directory for blocking page, and copying files"
     printf "  %b %s..." "${INFO}" "${str}"
     # Install the directory
-    install -d /var/www/html/pihole
+    install -d -m 0755 $PI_HOLE_DASHBOARD_DIR
```
Comment (dschaper) [inline automated install/basic-install.sh]: Please use the `${VAR}` form of expansion.

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -1343,8 +1351,9 @@ installConfigs() {
     # Make sure Pi-hole's config files are in place
     version_check_dnsmasq
     # Install empty file if it does not exist
-    if [[ ! -f "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" ]]; then
-        if ! install -o pihole -g pihole -m 664 /dev/null "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" &>/dev/null; then
+    if [[ ! -r "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" ]]; then
+        install -d -m 0755 ${PI_HOLE_CONFIG_DIR}
+        if ! install -o pihole -m 664 /dev/null "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" &>/dev/null; then
```
Comment (dschaper) [inline automated install/basic-install.sh]: How about `install -D` and then there's no need to have separate directory and file creation.

File: automated install/basic-install.sh (hunk 1 above)
Comment (cryptomilk) [reply automated install/basic-install.sh]: OK, lets check the content too, to be sure.

File: automated install/basic-install.sh (hunk 2 above)
Comment (cryptomilk) [reply automated install/basic-install.sh]: Ok, fixed.

File: automated install/basic-install.sh (hunk 3 above)
Comment (cryptomilk) [reply automated install/basic-install.sh]: But with a seperate command you can provide the permissions for the directory. With -D you don't really know what you end up. It depends on the default umask set.

Comment (cryptomilk) [reply automated install/basic-install.sh]: Lessons learned from RPM packaging ;-)

**Comment by cryptomilk** [comment]
Pushed new branch with requested changes.

**Comment by cryptomilk** [comment]
I did more test runs. Looks fine now. I have also more stuff in the queue :-)

**Comment by Mcat12** [comment]
We require PRs to be made based on the `development` branch, and to be merged into the `development` branch. This PR is based on `master` and is merging to `master.

**Comment by cryptomilk** [comment]
Another project redefining git defaults :-P

I will recreate it against development ...

**Comment by Mcat12** [comment]
@cryptomilk We use GitFlow, which is a very standard Git workflow: https://nvie.com/posts/a-successful-git-branching-model/
There is no "default" git workflow.

Also, you explicitly checked that you read the contributor's guide, which mentions in bold to only submit PRs to development.

**Pull request closed without merging**